### ISSUE TITLE
MSH segments are misparsed, preventing GetValue() selection

### ISSUE THREAD

**Issue by JeremiahSanders** [Efferent-Health/HL7-dotnetcore#26]
Upon parsing existing HL7 data using `HL7-dotnetcore` version `2.8.0` from NuGet, `MSH` segment fields cannot be accessed by their correct index. They suffer from an off-by-1 index issue. Specifically, it appears that `MSH.1` is being misparsed. This severely impacts querying `MSH` data.

*Expectation*

According to the spec ( https://www.hl7.org/documentcenter/public_temp_CACD15D9-1C23-BA17-0C050D19F5A35765/wg/conf/HL7MSH.htm ), `MSH.1` should be have a value of `|` for conventional messages and `MSH.2` should have a value of `^~\&`.

*Result*

Actual parsing of a very simple, faux message results in `MSH.1` getting parsed with the `MSH.2` value and all subsequent `MSH.#` queries are 1 index higher than they should be.

*Code to reproduce*

```c#
static void Test()
{
	// HL7 string is intentionally multi-line to ensure segment delimiter is recognized
	const string dummyA08 = @"MSH|^~\&|sendingApplication|sendingFacility|receivingApplication|receivingFacility|20100314063000||ADT^A08|-1|D||||
";
	var message = new Message(dummyA08);
	message.ParseMessage();
	Console.WriteLine($"MSH.1 - expected '|' - actual '{message.GetValue("MSH.1")}'");
	const string msh2Expectation = @"^~\&";
	Console.WriteLine($"MSH.2 - expected '{msh2Expectation}' - actual '{message.GetValue("MSH.2")}'");
	Console.WriteLine($"MSH.3 - expected 'sendingApplication' - actual '{message.GetValue("MSH.3")}'");
}
```

**Comment by jaime-olivares** [comment]
I will solve this very soon

**Comment by jaime-olivares** [comment]
Published as NuGet package 2.9.0
